engine/src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -498,16 +523,19 @@ class WorkflowActor(workflowToStart: WorkflowToStart,
           }
         }
       }
-      context stop self
-  }
 
-  onTransition {
+      // We can't transition from within another transition function, but we can instruct ourselves to with a message:
+      self ! AwaitMetadataIntegrity
+
+      // Push the final status and ask for an acknowledgement (which we will now be waiting for):
+      pushCurrentStateToMetadataService(workflowId, terminalState.workflowState, confirmTo = Option(self))
+
     case fromState -> toState =>
       workflowLogger.debug(s"transitioning from {} to {}", arg1 = fromState, arg2 = toState)
       // This updates the workflow status
       // Only publish "External" state to metadata service
       // workflowState maps a state to an "external" state (e.g all states extending WorkflowActorRunningState map to WorkflowRunning)
-      if (fromState.workflowState != toState.workflowState) {
+      if (fromState.workflowState != toState.workflowState && toState != MetadataIntegrityValidationState) {
```

Comment:
Isn't this a "programmer error" sort of scenario? If we try to write new metadata while we're flushing the metadata queue, it seems like the app violated an invariant and we should report it.

Reply:
This specific change is catching us moving away from (eg) the `WorkflowActorSucceededState` to the `MetadataIntegrityValidationState`, and saying "actually... don't record that as a state change in the database".

You're right that any metadata that comes in after the final status is probably a mistake, but I'm not sure there's an easy or simple way to catch that.